test/test_databases/test_postgres.py

```diff
@@ -0,0 +1,179 @@
+        # Test varchar that looks like an int
+        self.assertEqual(self.pg.data_type('00001', ''), 'varchar')
+        # Test a float as a decimal
+        self.assertEqual(self.pg.data_type(5.001, ''), 'decimal')
+        # Test varchar
+        self.assertEqual(self.pg.data_type('word', ''), 'varchar')
+        # Test int with underscore
+        self.assertEqual(self.pg.data_type('1_2', ''), 'varchar')
+        # Test int with leading zero
+        self.assertEqual(self.pg.data_type('01', ''), 'varchar')
+
+    def test_generate_data_types(self):
+
+        # Test correct header labels
+        self.assertEqual(self.mapping['headers'], ['ID', 'Name'])
+        # Test correct data types
+        self.assertEqual(self.mapping['type_list'], ['smallint', 'varchar'])
+        # Test correct lengths
+        self.assertEqual(self.mapping['longest'], [1, 5])
+
+    def test_vc_padding(self):
+
+        # Test padding calculated correctly
+        self.assertEqual(self.pg.vc_padding(self.mapping, .2), [1, 6])
+
+    def test_vc_max(self):
+
+        # Test max sets it to the max
+        self.assertEqual(self.pg.vc_max(self.mapping, ['Name']), [1, 65535])
+
+        # Test raises when can't find column
+        # To Do
+
+    def test_vc_validate(self):
+
+        # Test that a column with a width of 0 is set to 1
+        self.mapping['longest'][0] = 0
+        self.mapping = self.pg.vc_validate(self.mapping)
+        self.assertEqual(self.mapping, [1, 5])
+
+    def test_create_sql(self):
+
+        # Test the the statement is expected
+        sql = self.pg.create_sql('tmc.test', self.mapping, distkey='ID')
+        exp_sql = "create table tmc.test (\n  id smallint,\n  name varchar(5)) \ndistkey(ID) ;"
+        self.assertEqual(sql, exp_sql)
+
+    def test_column_validate(self):
+
+        bad_cols = ['', 'SELECT', 'asdfjkasjdfklasjdfklajskdfljaskldfjaklsdfjlaksdfjklasjdfklasjdkfljaskldfljkasjdkfasjlkdfjklasdfjklakjsfasjkdfljaslkdfjklasdfjklasjkldfakljsdfjalsdkfjklasjdfklasjdfklasdkljf'] # noqa: E501
+        fixed_cols = ['col_0', 'col_1', 'asdfjkasjdfklasjdfklajskdfljaskldfjaklsdfjlaksdfjklasjdfklasjdkfljaskldfljkasjdkfasjlkdfjklasdfjklakjsfasjkdfljaslkdfjkl'] # noqa: E501
+        self.assertEqual(self.pg.column_name_validate(bad_cols), fixed_cols)
+
+    def test_create_statement(self):
+
+        # Assert that copy statement is expected
+        sql = self.pg.create_statement(self.tbl, 'tmc.test', distkey='ID')
+        exp_sql = """create table tmc.test (\n  "id" smallint,\n  "name" varchar(5)) \ndistkey(ID) ;"""  # noqa: E501
+        self.assertEqual(sql, exp_sql)
+
+        # Assert that an error is raised by an empty table
+        empty_table = Table([['Col_1', 'Col_2']])
+        self.assertRaises(ValueError, self.pg.create_statement, empty_table, 'tmc.test')
+
+# These tests interact directly with the Postgres database
+
+@unittest.skipIf(not os.environ.get('LIVE_TEST'), 'Skipping because not running live test')
```

Comment:
Might be a good idea to add db testing to the CI workflow - https://discuss.circleci.com/t/how-can-i-start-testing-database-connection-in-circleci/21512

Reply:
Yes. That could be a good future PR.